### ISSUE TITLE
The library does not work correctly when used via `esm-cdn`

### ISSUE THREAD

**Issue by Qsppl** [amcharts/amcharts5#1797]
The library does not work correctly when used via esm-cdn

## Expected

See [Donut Chart - Working example](https://codepen.io/qsppl/pen/jENNoMV)

![image](https://github.com/user-attachments/assets/1d032609-9ae7-453d-aab7-7c7580328370)

## Observed

~~See [Donut Chart - Non-working example](https://codepen.io/qsppl/pen/KwPPLaa)~~ (deprecated)
See [Donut Chart - Non-working example](https://codepen.io/qsppl/pen/GgKoPjJ)

![image](https://github.com/user-attachments/assets/96edffe9-bcc5-4184-a089-e4d2acb84e29)

## Additional info

I have raised a related issue in the esm.sh repository https://github.com/esm-dev/esm.sh/issues/934

Could you share more details for `ESM.SH` developers? What exactly was assembled incorrectly? Is there a bug somewhere in the code or is some specific dependency not working correctly?

I also see that this issue has already been raised previously in https://github.com/amcharts/amcharts5/issues/1564, but I was unable to continue that discussion.

**Comment by Pauan** [comment]
I've looked at the build output of esm.sh... I did notice a few weird things.

For some strange reason it uses `.mjs` for most of the files, but for `Animated` and `percent` it uses `.js`

I also noticed that it's exporting twice from `Animated.js`:

```js
/* esm.sh - @amcharts/amcharts5@5.10.8/themes/Animated */
export * from "/v135/@amcharts/amcharts5@5.10.8/es2022/themes/Animated.js";
export { default } from "/v135/@amcharts/amcharts5@5.10.8/es2022/themes/Animated.js";
```

It shouldn't need the second `export { default } from "...";`

And it's importing `tslib` and `d3-shape` multiple times, when it shouldn't:

```js
/* esm.sh - @amcharts/amcharts5@5.10.8 */
import "/v135/tslib@2.8.1/es2022/tslib.mjs";
import "/v135/svg-arc-to-cubic-bezier@3.2.0/es2022/svg-arc-to-cubic-bezier.mjs";
import "/v135/d3-shape@3.2.0/es2022/d3-shape.mjs";
export * from "/v135/@amcharts/amcharts5@5.10.8/es2022/amcharts5.mjs";
```

```js
/* esm.sh - @amcharts/amcharts5@5.10.8/percent */
import "/v135/tslib@2.8.1/es2022/tslib.mjs";
import "/v135/d3-shape@3.2.0/es2022/d3-shape.mjs";
export * from "/v135/@amcharts/amcharts5@5.10.8/es2022/percent.js";
```

The above things are pretty minor, but I found a *major* bug with esm.sh.

I noticed that it's not handling cross-file imports properly. The `percent.js` and `Animated.js` files are supposed to be importing things from `amcharts5.mjs`... but they aren't. There's no imports at all.

So it seems to be splitting each file into separate unrelated chunks, instead of sharing a single chunk.

**Comment by Qsppl** [comment]
Thank you very much for the work done, I will double-check each error and open relevant issues in the ESM.SH repository.

Please report if you find any more errors.

I will report progress on the issues raised in this thread so that you can return to this issue at a later date.

**Comment by Qsppl** [comment]
In the broken chart example, I accidentally used multiple versions of AMCharts.

![image](https://github.com/user-attachments/assets/19cec95e-ae5c-475a-93dc-6d538105f972)

In the past, this might have prevented you from finding the real problem.

I have fixed this problem and updated the example link in the first post of the thread.

![image](https://github.com/user-attachments/assets/e464169d-15d0-4183-b60f-fdddd80e0cc6)

Just in case, I’ll write down here a link to the previous example with importing several versions of AMCharts, so that you can compare the examples if necessary: https://codepen.io/qsppl/pen/KwPPLaa

**Comment by Qsppl** [comment]
> I noticed that it's not handling cross-file imports properly. The percent.js and Animated.js files are supposed to be importing things from amcharts5.mjs... but they aren't. There's no imports at all.

> So it seems to be splitting each file into separate unrelated chunks, instead of sharing a single chunk.

Here in the file five `imports` of different modules of one project are performed:

![image](https://github.com/user-attachments/assets/5e698e03-10cb-45a8-8d8a-f26cab5cd669)

But it seems that ESM.SH returned not five modules of one project, but five differently builded projects.

![image](https://github.com/user-attachments/assets/d4811813-c271-4681-9c7d-de8ff3a99264)

Accordingly, all variables exported from modules are their own for each `import`.

This must be the problem I was looking for. Thank you very much.

**Comment by Pauan** [comment]
> But it seems that ESM.SH returned not five modules of one project, but five differently builded projects.

Yes, that's what I meant when I said that it's building multiple chunks instead of sharing a single chunk.

**Comment by Qsppl** [comment]
I don’t think that further solving the problem will require the active participation of AmCharts developers, so I’m closing the question.

Thank you very much for responding and identifying the problem, I hope the issue will be resolved soon and AmCharts will become available as an ESM module downloadable from a CDN!

**Comment by Qsppl** [comment]
Hi, I found a solution to the problem with importing AMCharts via CDN from multiple entry points, just use `?bundle=false`:

```js
import * as am5 from 'https://esm.sh/@amcharts/amcharts5@5.10.8?bundle=false';
import * as am5xy from 'https://esm.sh/@amcharts/amcharts5@5.10.8/xy?bundle=false';
import * as am5index from "https://esm.sh/@amcharts/amcharts5@5.10.8/index?bundle=false";
import * as am5percent from "https://esm.sh/@amcharts/amcharts5@5.10.8/percent?bundle=false";
import am5themes_Animated from "https://esm.sh/@amcharts/amcharts5@5.10.8/themes/Animated?bundle=false";
```

[...full example](https://codepen.io/qsppl/pen/wBvmgKQ)

You can describe this way of importing a package in the documentation, it is fully working and does not create problems with performance on HTTP2 and above. Good luck!
